tedge_config/src/tedge_config.rs

```diff
@@ -97,13 +80,12 @@ impl ConfigSettingAccessor<C8yUrlSetting> for TEdgeConfig {
 
 impl ConfigSettingAccessor<DeviceCertPathSetting> for TEdgeConfig {
     fn query(&self, _setting: DeviceCertPathSetting) -> ConfigSettingResult<String> {
-        self.data
+        Ok(self
+            .data
             .device
             .cert_path
             .clone()
-            .ok_or(ConfigSettingError::ConfigNotSet {
-                key: DeviceCertPathSetting::KEY,
-            })
+            .unwrap_or_else(|| DEFAULT_DEVICE_CERT_PATH.into()))
```

Comment:
This is closer to what I was expecting (i.e the default value along the rule). But this works only for tedge used at the system level (i.e `sudo tedge`).

* When running as a regular user, the default is different and depends of $HOME.